chain/errors.go

```diff
@@ -0,0 +1,495 @@
+// - https://github.com/bitcoin/bitcoin/blob/master/test/functional/mempool_limit.py
+// - https://github.com/bitcoin/bitcoin/blob/master/src/validation.cpp
+func (r RPCErr) Error() string {
+	switch r {
+	case ErrMissingInputsOrSpent:
+		return "bad txns inputs missingorspent"
+
+	case ErrMaxBurnExceeded:
+		return "Unspendable output exceeds maximum configured by user (maxburnamount)"
+
+	case ErrMaxFeeExceeded:
+		return "max fee exceeded"
+
+	case ErrTxAlreadyKnown:
+		return "txn already known"
+
+	case ErrTxAlreadyConfirmed:
+		return "Transaction already in block chain"
+
+	case ErrMempoolConflict:
+		return "txn mempool conflict"
+
+	case ErrReplacementAddsUnconfirmed:
+		return "replacement adds unconfirmed"
+
+	case ErrInsufficientFee:
+		return "insufficient fee"
+
+	case ErrTooManyReplacements:
+		return "too many potential replacements"
+
+	case ErrMempoolMinFeeNotMet:
+		return "mempool min fee not met"
+
+	case ErrConflictingTx:
+		return "bad txns spends conflicting tx"
+
+	case ErrEmptyOutput:
+		return "bad txns vout empty"
+
+	case ErrEmptyInput:
+		return "bad txns vin empty"
+
+	case ErrTxTooSmall:
+		return "tx size small"
+
+	case ErrDuplicateInput:
+		return "bad txns inputs duplicate"
+
+	case ErrEmptyPrevOut:
+		return "bad txns prevout null"
+
+	case ErrBelowOutValue:
+		return "bad txns in belowout"
+
+	case ErrNegativeOutput:
+		return "bad txns vout negative"
+
+	case ErrLargeOutput:
+		return "bad txns vout toolarge"
+
+	case ErrLargeTotalOutput:
+		return "bad txns txouttotal toolarge"
+
+	case ErrScriptVerifyFlag:
+		return "mandatory script verify flag failed"
+
+	case ErrTooManySigOps:
+		return "bad txns too many sigops"
+
+	case ErrInvalidOpcode:
+		return "disabled opcode"
+
+	case ErrTxAlreadyInMempool:
+		return "txn already in mempool"
+
+	case ErrMissingInputs:
+		return "missing inputs"
+
+	case ErrOversizeTx:
+		return "bad txns oversize"
+
+	case ErrCoinbaseTx:
+		return "coinbase"
+
+	case ErrNonStandardVersion:
+		return "version"
+
+	case ErrNonStandardScript:
+		return "scriptpubkey"
+
+	case ErrBareMultiSig:
+		return "bare multisig"
+
+	case ErrScriptSigNotPushOnly:
+		return "scriptsig not pushonly"
+
+	case ErrScriptSigSize:
+		return "scriptsig size"
+
+	case ErrTxTooLarge:
+		return "tx size"
+
+	case ErrDust:
+		return "dust"
+
+	case ErrMultiOpReturn:
+		return "multi op return"
+
+	case ErrNonFinal:
+		return "non final"
+
+	case ErrNonBIP68Final:
+		return "non BIP68 final"
+
+	case ErrSameNonWitnessData:
+		return "txn same nonwitness data in mempool"
+
+	case ErrNonMandatoryScriptVerifyFlag:
+		return "non mandatory script verify flag"
+	}
+
+	return "unknown error"
+}
+
+// BtcdErrMap takes the errors returned from btcd's `testmempoolaccept` and
+// `sendrawtransaction` RPCs and map them to the errors defined above, which
+// are results from calling either `testmempoolaccept` or `sendrawtransaction`
+// in `bitcoind`.
+//
+// Errors not mapped in `btcd`:
```

Comment:
Q: Do you mean these are particular btcd errros which exist only in btcd or the other way around ?

Reply:
so these errors can be returned from `btcd`, but they are not mapped. so yeah they only exist in `btcd`.